.github/workflows/main.yml

```diff
@@ -0,0 +1,132 @@
+        - os: macos-latest
+          rust: stable
+          other: x86_64-apple-ios
+        - os: windows-latest
+          rust: stable-msvc
+          other: i686-pc-windows-msvc
+        - os: windows-latest
+          rust: beta-gnu
+          other: i686-pc-windows-gnu
+    steps:
+    - uses: actions/checkout@v2
+    - run: rustup update ${{ matrix.rust }} && rustup default ${{ matrix.rust }}
+    - run: rustup target add ${{ matrix.other }}
+    - run: rustup component add rustc-dev llvm-tools-preview rust-docs
+      if: matrix.rust == 'nightly'
+    - run: sudo apt update -y && sudo apt install gcc-multilib -y
+      if: matrix.os == 'ubuntu-latest'
+    - run: rustup component add rustfmt || echo "rustfmt not available"
+
+    # Deny warnings on CI to avoid warnings getting into the codebase, and note
+    # the `force-system-lib-on-osx` which is intended to fix compile issues on
+    # OSX where compiling curl from source on OSX yields linker errors on Azure.
+    #
+    # Note that the curl issue is traced back to alexcrichton/curl-rust#279
+    # where it looks like the OSX version we're actually running on is such that
+    # a symbol is emitted that's never worked. For now force the system library
+    # to be used to fix the link errors.
+    - run: cargo test --features 'deny-warnings curl/force-system-lib-on-osx'
+    - run: cargo test -p cargo-test-support
+    - run: cargo test -p cargo-platform
+
+  resolver:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - run: rustup update stable && rustup default stable
+    - run: cargo test --manifest-path crates/resolver-tests/Cargo.toml
+
+  build_std:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - run: rustup update nightly && rustup default nightly
+    - run: cargo test --manifest-path crates/resolver-tests/Cargo.toml
+    - run: rustup component add rust-src
+    - run: cargo build
+    - run: cargo test --test build-std
+      env:
+        CARGO_RUN_BUILD_STD_TESTS: 1
+  docs:
+    env:
+      MDBOOK_VERSION: 0.3.7
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - run: rustup update nightly && rustup default nightly
+    - run: rustup component add rust-docs
+
+    # Install mdbook, cached
+    - uses: actions/cache@v1
+      with:
+        path: ${{ runner.tool_cache }}/mdbook
+        key: mdbook-${{ env.MDBOOK_VERSION }}
+    - run: echo "::add-path::${{ runner.tool_cache }}/mdbook/bin"
+    - run: cargo install --root ${{ runner.tool_cache }}/mdbook --version ${{ env.MDBOOK_VERSION }} mdbook
+
+    # Build docs
+    - run: cargo doc --no-deps
+    - run: cd src/doc && mdbook build --dest-dir ../../target/doc
+
+    # Run a simple linkchecker
+    - run: |
+        cd src/doc
+        curl -sSLo linkcheck.sh \
+          https://raw.githubusercontent.com/rust-lang/rust/master/src/tools/linkchecker/linkcheck.sh
+        sh linkcheck.sh --all cargo
+
+  success:
```

Comment:
This is some trickery here to get bors to work. Bors can only listen for one status on a PR, but GitHub Actions creates a status-per-builder. This "bors build finished" status is intended to be the signal to bors that the build has finished. This is mostly just copied from rust-lang/rust. I'm hoping that bors-rs may make this a bit nicer so we don't have to do this in the future.